pkg/jx/cmd/step_create_task.go

```diff
@@ -392,74 +401,130 @@ func (o *StepCreateTaskOptions) generatePipeline(languageName string, pipelineCo
 	return o.writeTask(fileName, task)
 }
 
-func (o *StepCreateTaskOptions) writeTask(fileName string, task *pipelineapi.Task) error {
-	data, err := yaml.Marshal(task)
+func (o *StepCreateTaskOptions) generatePipelineFromYaml(yamlFile string) error {
+	yamlBytes, err := ioutil.ReadFile(yamlFile)
 	if err != nil {
-		return errors.Wrapf(err, "failed to marshal Task YAML")
+		return errors.Wrapf(err, "Failed to read YAML file: %s", yamlFile)
 	}
-	err = ioutil.WriteFile(fileName, data, util.DefaultWritePermissions)
+	yamlString := string(yamlBytes) // TODO: What encoding does this use?
+	parsed, err := buildpipeline.ParseJenkinsfileYaml(yamlString)
 	if err != nil {
-		return errors.Wrapf(err, "failed to save Task file %s", fileName)
+		return errors.Wrapf(err, "Failed to parse YAML for: %s", yamlFile)
+	}
+	validateErr := parsed.Validate()
+	if validateErr != nil {
+		return errors.Wrapf(validateErr, "Validation failed for: %s", yamlFile)
+	}
+	o.setBuildValues()
+	// TODO: use org-name-branch for pipeline name? Create client now to get
+	// namespace? Set namespace when applying rather than during generation?
+	pipeline, tasks, err := parsed.GenerateCRDs("somepipeline", o.buildNumber, "ns", "abcd")
+	// TODO: where should this be created? In GenerateCRDs?
```

Comment:
Good question. Need to figure out in general where the CRD generation goes - in `buildpipeline` or in `kpipelines`...probably in `kpipelines`, since that's already there and is thematically the same as `buildpipeline`. But we can leave it as is for the moment and refactor in followups.